components/eamxx/docs/developer/processes.md

```diff
@@ -1,18 +1,210 @@
+and easy to test via a battery of targeted unit tests.
+
+In EAMxx, we already have a few concrete AP's, interfacing the AD to the Hommexx non-hydrostatic dycore as well as
+some physics parametrizations (P3, SHOC, RRMTPG, etc). In the next section we describe the interfaces of an AP class,
+and we show an example of how to write a new concrete AP class.
+
+## Atmosphere process interfaces
+
+An AP has several interfaces, which can be grouped into three categories:
+
+ - initialization: these interfaces are used to create the AP, as well as to initialize internal data structures;
+ - run: these interfaces are used to make the AP compute its output fields from its input fields;
+ - finalization: these interfaces are used to perform any clean up operation (e.g., release files) before the AP is
+ destroyed.
+
+Among the above, the initialization sequence is the most complex, and conists of several steps:
+
+ - The AD creates the APG corresponding to the whole atmosphere. As mentioned above, this phase will make use of a factory,
+   which allows the AD to be agnostic to what is actually in the group. All AP's can start performing any initialization
+   work that they can, but at this point they are limited to use only an MPI communicator as well as a list of runtime
+   parameters (which were previously read from an input file).
+ - The AD passes a `GridsManager` to the AP's, so that they can get information about the grids they need. At this point,
+   all AP's have all the information they need to establish the layout of the input and output fields they need,
+   and can store a list of these "requests"
+ - After creating all fields (based on AP's requests), the AD passes a copy of each input and output field to
+   the AP's. These fields will be divided in "required" and "computed", which differ in that the former are only
+   passed to the AP's as 'read-only' fields (see the [field](field.md#Field) documentation for more details)
+ - The AP's are queried for how much scratch memory they may need at run time. After all AP's communicate their needs,
+   the AD will provide a pointer to scratch memory to the AP's. This is memory that can be used to initialize
+   temporary views/fields or other internal data structures. All AP's are given the same pointer, which means no
+   data persistence should be expected at run time between one timestep and the next.
+ - The AD calls the 'initialize' method on each AP. At this point, all fields are set, and AP's can complete any
+   remaining initialization task
+
+While the base AP class provides an (empty) implementation for some methods, in case derived classes do not need a
+feature, some methods are purely virtual, and concrete classes will have to override them. Looking at existing
+concrete AP implementations is a good way to have a first idea of what a new AP class needs to implement. Here,
+we show go over the possible implementation of these methods in a hypothetical AP class. The header file may
```

Comment:
Does markdown allow us to boldface text?  If so then we should boldface "header file".

Reply:
It does. Why do you think it helps to be bold face?

Reply:
Just to avoid confusion for new-to-C++ developers.  They will be learning about the difference between `.hpp` and `.cpp` so emphasizing that this is an example for a `.hpp` file may be helpful for them.

Reply:
I'm fine without bold-face though.  I'll approve and let you decide if you want to make the change.